test/extended/builds/digest.go

```diff
@@ -75,8 +77,15 @@ func testBuildDigest(oc *exutil.CLI, buildFixture string, buildLogLevel uint) {
 		g.By("checking that the image digest has been saved to the build status")
 		o.Expect(br.Build.Status.Output.To).NotTo(o.BeNil())
 
-		ist, err := oc.ImageClient().ImageV1().ImageStreamTags(oc.Namespace()).Get(context.Background(), "test:latest", v1.GetOptions{})
+		ist, err := oc.ImageClient().ImageV1().ImageStreamTags(oc.Namespace()).Get(context.Background(), "test:latest", metav1.GetOptions{})
 		o.Expect(err).NotTo(o.HaveOccurred())
 		o.Expect(br.Build.Status.Output.To.ImageDigest).To(o.Equal(ist.Image.Name))
+
+		g.By("checking that the image layers have valid docker v2schema2 MIME types")
+		image, err := oc.ImageClient().ImageV1().Images().Get(context.Background(), br.Build.Status.Output.To.ImageDigest, metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		for _, layer := range image.DockerImageLayers {
+			o.Expect(layer.MediaType).To(o.Equal(manifestschema2.MediaTypeLayer))
```

Comment:
@nalind per our discussion in Slack, we are requiring all pushed layers to have the default v2s2 MIME type (`application/vnd.docker.image.rootfs.diff.tar.gzip`)

Reply:
Yep, if we know it's v2s2, I think that's correct.  Given that we generated it, I don't think we need to consider MediaTypeForeignLayer.